### PR TITLE
Allow pandoc 3

### PR DESCRIPTION
**Pull request by LaurentRDC**
This is a simple version bump to allow pandoc-3.

I've tried my fork with [my own website](https://github.com/LaurentRDC/personal-website) and everything is working as expected.

### CODE REVIEW

**Comment by LaurentRDC** [comment]
I had to update some of the golden tests because pandoc-3's output is different. I'm assuming that pandoc 3 is correct, but I can't really say.

File: hakyll.cabal [hunk 1]
```diff
@@ -246,7 +248,7 @@ Library
     Other-Modules:
       Hakyll.Web.Pandoc.Binary
     Build-Depends:
-      pandoc >= 2.11 && < 2.20
+      pandoc >= 2.11 && < 3.1
```
Comment (Minoru) [inline hakyll.cabal:251]: I suggest this is changed to a more restrictive clause similar to the ones for `aeson` and `text`:

```suggestion
      pandoc >= 2.11 && < 2.19.2 || >= 3.0 && < 3.1
```

Ditto in two places below.

File: hakyll.cabal [hunk 2]
```diff
@@ -208,13 +210,13 @@ Library
     time-locale-compat   >= 0.1      && < 0.2,
     unordered-containers >= 0.2      && < 0.3,
     vector               >= 0.11     && < 0.14,
+    wai-app-static       >= 3.1      && < 3.2,
```
Comment (Minoru) [inline hakyll.cabal:213]: Why was this moved from under the `previewServer` flag? I build current `master` (f466be79f58b34fd94f2249d2731a31c67a2a26d) with the following command, and it works fine:

```
$ for action in build test ; do cabal $action --enable-tests --flags='-previewServer' || break ; done
```

**Review by Minoru — APPROVED** [review]
Thanks! This looks good, but I'd like to know why the `wai-app-static` dependency was moved before I merge this.

The change in Pandoc's output is a bit unfortunate, because Hakyll's users might depend on these details in,say, their CSS. Can you please add a stub to the CHANGELOG so I don't forget to mention this when I do a release?

**Comment by Minoru** [comment]
Note to self: after merging this, make a new Hackage revision and include the changes from https://github.com/jaspervdj/hakyll/pull/961 in it, because I forgot to do so last time.

File: hakyll.cabal (hunk 2 above)
Comment (LaurentRDC) [reply hakyll.cabal:213]: That's very strange, master isn't building for me:

```bash
lib/Hakyll/Core/Configuration.hs:14:1: error:
    Could not load module ‘Network.Wai.Application.Static’
    It is a member of the hidden package ‘wai-app-static-3.1.7.4’.
    Perhaps you need to add ‘wai-app-static’ to the build-depends in your .cabal file.
    Use -v (or `:set -v` in ghci) to see a list of the files searched for.
   |
14 | import qualified Network.Wai.Application.Static as Static
```

using ghc 9.4.4

Comment (Minoru) [reply hakyll.cabal:213]: Oh, sorry, I was building your branch >_< It does fail on master for me too.

**Comment by Minoru** [comment]
Thank you!

**Comment by Minoru** [comment]
> Note to self: after merging this, make a new Hackage revision and include the changes from https://github.com/jaspervdj/hakyll/pull/961 in it, because I forgot to do so last time. Also, PR mtl 2.3 support that I have on a private branch.

No, wait, since it needed some adjustments to tests, I'd have to make a full-fledged release for this. I hope to do it this weekend.

**Comment by ozkutuk** [comment]
Is there a particular reason why the upper constraint for Pandoc 2 has changed from `< 2.20` to `< 2.19.2`? I depend on Nix for the CI builds of my blog and Nix provides `pandoc == 2.19.2`, so I can't build the current master of Hakyll from source with the given boundaries. Would it be a good idea to relax the upper bound on Pandoc 2 back to `< 2.20`?

**Comment by LaurentRDC** [comment]
Ahh this is my mistake. I wanted to write <= 2.19.2 (the last released version), but it could also be <= 2.20.

@Minoru Can you fix this on my behalf?

**Comment by Minoru** [comment]
Done in 909e1b3a89b5b3ba5f64840d23ada9b3ac393404. Sorry I led you down the wrong way, it was my suggestion to use < 2.19.2.
